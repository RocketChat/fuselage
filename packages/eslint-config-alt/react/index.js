module.exports = {
  extends: '../minimal',
  plugins: ['react', 'react-hooks'],
  parser: '@babel/eslint-parser',
  rules: {
    'jsx-quotes': ['error', 'prefer-single'],
    'react/display-name': 'error',
    'react/jsx-uses-react': 'error',
    'react/jsx-uses-vars': 'error',
    'react/jsx-no-undef': 'error',
    'react/jsx-fragments': ['error', 'syntax'],
    'react/no-multi-comp': 'error',
    'react/react-in-jsx-scope': 'error',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'warn',
  },
  settings: {
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
    },
    'react': {
      version: 'detect',
    },
  },
  env: {
    browser: true,
    es6: true,
  },
  overrides: [
    {
      files: ['**/*.ts', '**/*.tsx'],
      extends: '../typescript',
      parser: '@typescript-eslint/parser',
      plugins: ['react', 'react-hooks'],
      rules: {
        'jsx-quotes': ['error', 'prefer-single'],
        'react/display-name': 'error',
        'react/jsx-uses-react': 'error',
        'react/jsx-uses-vars': 'error',
        'react/jsx-no-undef': 'error',
        'react/jsx-fragments': ['error', 'syntax'],
        'react/no-multi-comp': 'error',
        'react/react-in-jsx-scope': 'error',
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'warn',
      },
      env: {
        browser: true,
        es6: true,
      },
      settings: {
        'import/resolver': {
          node: {
            extensions: ['.js', '.jsx', '.ts', '.tsx'],
          },
        },
        'react': {
          version: 'detect',
        },
      },
    },
    {
      files: [
        '**/*.stories.js',
        '**/*.stories.jsx',
        '**/*.stories.ts',
        '**/*.stories.tsx',
      ],
      rules: {
        'react/display-name': 'off',
        'react/no-multi-comp': 'off',
      },
    },
  ],
};
