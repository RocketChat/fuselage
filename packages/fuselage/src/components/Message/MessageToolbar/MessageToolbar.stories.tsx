import { Title, Primary } from '@storybook/addon-docs';
import type { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';

import type { MessageToolbarWrapper } from '.';
import { MessageToolbar, MessageToolbarItem } from '.';
import { Box } from '../..';

export default {
  title: 'Message/MessageToolbar',
  component: MessageToolbar,
  parameters: {
    docs: {
      page: () => (
        <>
          <Title />
          <Primary />
        </>
      ),
    },
  },
} as ComponentMeta<typeof MessageToolbarWrapper>;

export const Default: ComponentStory<typeof MessageToolbar> = () => (
  <Box>
    <MessageToolbar>
      <MessageToolbarItem title='Quote' icon='quote' />
      <MessageToolbarItem title='Emoji' icon='emoji' />
      <MessageToolbarItem title='Thread' icon='thread' />
      <MessageToolbarItem title='Menu' icon='menu' />
    </MessageToolbar>
  </Box>
);
