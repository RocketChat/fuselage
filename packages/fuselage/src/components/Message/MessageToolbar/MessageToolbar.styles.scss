@use '../../../styles/lengths.scss';
@use '../../../styles/colors.scss';
@use '../../../styles/typography.scss';
@use '../mixins.scss';

.rcx-message-toolbar {
  display: inline-block;

  margin-inline: lengths.margin(20);
  padding: lengths.padding(2);

  opacity: 0;

  border: lengths.border-width(default) solid colors.stroke(extra-light);
  border-radius: theme(
    'message-toolbar-border-radius',
    lengths.border-radius(medium)
  );
  background: colors.surface(room);

  &__wrapper {
    display: none;

    .rcx-message:hover &,
    .rcx-message:focus-within & {
      display: inline-block;
    }

    &--visible {
      display: inline-block;

      .rcx-message-toolbar {
        opacity: 1;
      }
    }
  }

  .rcx-message:hover &,
  .rcx-message:has(:focus-visible) &,
  .rcx-message:focus-visible & {
    opacity: 1;
  }

  @at-root .rcx-message & {
    position: absolute;
    z-index: 10;
    top: lengths.margin(-24);
    right: 0;

    &:dir(rtl) {
      left: 0;
      right: initial;
    }
  }
}
