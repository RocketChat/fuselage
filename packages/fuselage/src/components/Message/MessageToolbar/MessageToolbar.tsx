import type { ComponentProps, ForwardedRef } from 'react';
import React, { forwardRef } from 'react';

import { ButtonGroup } from '../..';

type MessageToolbarProps = ComponentProps<typeof ButtonGroup>;

export const MessageToolbar = forwardRef(function MessageToolbar(
  props: MessageToolbarProps,
  ref: ForwardedRef<HTMLDivElement>
) {
  return (
    <div className='rcx-box rcx-box--full rcx-message-toolbar'>
      <ButtonGroup role='toolbar' ref={ref} small {...props} />
    </div>
  );
});
