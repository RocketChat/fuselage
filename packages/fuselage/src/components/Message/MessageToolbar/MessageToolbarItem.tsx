import type { ComponentProps, Ref } from 'react';
import React, { forwardRef } from 'react';

import { IconButton } from '../../Button';

type MessageToolbarItemProps = ComponentProps<typeof IconButton>;

export const MessageToolbarItem = forwardRef(function MessageToolbarItem(
  props: MessageToolbarItemProps,
  ref: Ref<HTMLButtonElement>
) {
  return <IconButton ref={ref} small {...props} />;
});
