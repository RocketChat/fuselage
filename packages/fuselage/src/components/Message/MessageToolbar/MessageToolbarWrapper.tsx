import type { HTMLAttributes, Ref } from 'react';
import React, { forwardRef } from 'react';

type MessageToolbarWrapperProps = HTMLAttributes<HTMLDivElement> & {
  visible?: boolean;
};

export const MessageToolbarWrapper = forwardRef(function MessageToolbarWrapper(
  { className, visible, ...props }: MessageToolbarWrapperProps,
  ref: Ref<HTMLDivElement>
) {
  return (
    <div
      ref={ref}
      className={[
        'rcx-box rcx-box--full rcx-message-toolbar__wrapper',
        visible && 'rcx-message-toolbar__wrapper--visible',
        className,
      ]
        .filter(Boolean)
        .join(' ')}
      {...props}
    />
  );
});
