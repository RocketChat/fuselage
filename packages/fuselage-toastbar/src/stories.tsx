import { Button, Box } from '@rocket.chat/fuselage';
import type { Meta, Story } from '@storybook/react';
import { useState } from 'react';

import { useToastBarDispatch } from './ToastBarContext';

export default {
  title: 'view/ToastBar',
  parameters: {
    layout: 'fullscreen',
    actions: { argTypesRegex: '^on.*' },
  },
} as Meta;

export const ToastBarWithData: Story = () => {
  const [counter, setCounter] = useState(0);
  const dispatchToastMessage = useToastBarDispatch();

  const messageArray = [
    'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ipsam nihi Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ipsam nihi',
    'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ipsam nihi Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ipsam nihi',
    'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ipsam nihi Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ipsam nihi',
    'Lorem Ipsum',
    'Lorem ipsum dolor, sit amet consectetur adipisicing elit.',
  ];

  const handleToast = () => {
    dispatchToastMessage({
      type: 'success',
      message: messageArray[counter],
    });

    dispatchToastMessage({
      type: 'error',
      message: messageArray[counter],
      time: 10,
      position: 'bottom-start',
    });

    if (counter === messageArray.length - 1) {
      return setCounter(0);
    }

    return setCounter((prevState) => prevState + 1);
  };

  return (
    <Box
      display='flex'
      alignItems='center'
      justifyContent='center'
      height='100vh'
    >
      <Button primary onClick={handleToast}>
        Dispatch ToastBar
      </Button>
    </Box>
  );
};
