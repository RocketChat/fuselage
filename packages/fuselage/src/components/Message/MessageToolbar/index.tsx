import { Menu } from '../../Menu';
import { MessageToolbar } from './MessageToolbar';
import { MessageToolbarItem } from './MessageToolbarItem';
import { MessageToolbarWrapper } from './MessageToolbarWrapper';

export default Object.assign(MessageToolbar, {
  /**
   * @deprecated prefer using named imports
   * */
  Item: MessageToolbarItem,
  /**
   * @deprecated prefer using named imports
   * */
  Wrapper: MessageToolbarWrapper,
  /**
   * @deprecated prefer using named imports
   * */
  Menu,
});

export { MessageToolbar, MessageToolbarItem, MessageToolbarWrapper, Menu };
