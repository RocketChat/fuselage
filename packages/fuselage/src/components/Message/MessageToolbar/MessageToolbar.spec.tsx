import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import * as stories from './MessageToolbar.stories';

const testCases = Object.values(composeStories(stories)).map((Story) => [
  Story.storyName || 'Story',
  Story,
]);

describe('[MessageToolbar Rendering]', () => {
  test.each(testCases)(
    `renders %s without crashing`,
    async (_storyname, Story) => {
      const tree = render(<Story />, { legacyRoot: true });
      expect(tree.baseElement).toMatchSnapshot();
    }
  );

  test.each(testCases)(
    '%s should have no a11y violations',
    async (_storyname, Story) => {
      const { container } = render(<Story />, { legacyRoot: true });

      const results = await axe(container);
      expect(results).toHaveNoViolations();
    }
  );
});
