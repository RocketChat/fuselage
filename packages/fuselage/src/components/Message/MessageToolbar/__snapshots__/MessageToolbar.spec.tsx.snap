// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MessageToolbar Rendering] renders Default without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full"
    >
      <div
        class="rcx-box rcx-box--full rcx-message-toolbar"
      >
        <div
          class="rcx-button-group rcx-button-group--align-start rcx-button-group--small"
          role="toolbar"
        >
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
            title="Quote"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-quote rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
            title="Emoji"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-emoji rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
            title="Thread"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-thread rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button  rcx-button-group__item"
            title="Menu"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-menu rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
